lib/impress.client.js

```diff
@@ -966,13 +983,12 @@ Client.prototype.compress = function(
 
   const sinceTime = client.req.headers['if-modified-since'];
   if (sinceTime && api.common.isTimeEqual(sinceTime, stats.mtime)) {
-    client.error(304);
-  } else {
-    application.compress(filePath, stats, (err, data, compressed) => {
-      if (err) client.error(404);
-      else client.end({ stats, compressed, data });
-    });
+    return client.error(304);
   }
+  application.compress(filePath, stats, (err, data, compressed) => {
+    if (err) return client.error(404);
```

Comment:
If compression can fail too, I'd like it to be logged and, probably, server should respond with 500.